main.ml

```diff
@@ -511,9 +504,10 @@ let process root_dir name =
         | Some "xeno+shm" ->
           let uid = Uri.path uri in
           let uid = if String.length uid > 1 then String.sub uid ~pos:1 ~len:(String.length uid - 1) else uid in
-          RRD.Client.Plugin.Local.register ~uid ~info:Rrd.Five_Seconds ~protocol:Rrd_interface.V2
-          >>= fun _ ->
-          loop (uid :: acc) datasources
+          (RRD.Client.Plugin.Local.register RRD.Client.rpc uid Rrd.Five_Seconds Rrd_interface.V2).Rpc_async.M.async
```

Comment:
Do you actually need `RRD.Client.rpc`? You are using just `RRD.rpc` at line 547. I think that's the right invocation of this call

Reply:
Turns out it does not!